pkg/action/failover.go

```diff
@@ -0,0 +1,384 @@
+		}
+		action.Status.FinishTimestamp = metav1.Now()
+		ac.updateAction(action)
+		return
+	}
+
+	var config *rest.Config
+	namespaces := action.Spec.ActionParameter.FailoverParameter.FailoverNamespaces
+
+	migrationSchedule, err := storkops.Instance().GetMigrationSchedule(action.Spec.ActionParameter.FailoverParameter.MigrationScheduleReference, action.Namespace)
+	if err != nil {
+		msg := fmt.Sprintf("error fetching migrationschedule %s/%s", action.Namespace, action.Spec.ActionParameter.FailoverParameter.MigrationScheduleReference)
+		logEvents := ac.printFunc(action, string(storkv1.ActionStatusFailed))
+		logEvents(msg, "err")
+		action.Status.Status = storkv1.ActionStatusFailed
+		action.Status.Reason = msg
+		ac.updateAction(action)
+		return
+	}
+
+	migrationNamespaces, err := utils.GetMergedNamespacesWithLabelSelector(migrationSchedule.Spec.Template.Spec.Namespaces, migrationSchedule.Spec.Template.Spec.NamespaceSelectors)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to get list of namespaces from migrationschedule %s/%s", migrationSchedule.Namespace, migrationSchedule.Name)
+		log.ActionLog(action).Errorf(msg)
+		ac.recorder.Event(action,
+			v1.EventTypeWarning,
+			string(storkv1.ActionStatusScheduled),
+			msg)
+		ac.updateAction(action)
+		return
+	}
+
+	if reactivate {
+		// in case of reactivation you have to activate the apps which were scaled down in the source cluster i.e. remote cluster
+		config, err = getClusterPairSchedulerConfig(migrationSchedule.Spec.Template.Spec.ClusterPair, migrationSchedule.Namespace)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get the remote config from the clusterpair %s", migrationSchedule.Spec.Template.Spec.ClusterPair)
+			logEvents := ac.printFunc(action, string(storkv1.ActionStatusFailed))
+			logEvents(msg, "err")
+			action.Status.Status = storkv1.ActionStatusFailed
+			action.Status.Reason = msg
+			ac.updateAction(action)
+			return
+		}
+	} else {
+		config = ac.config
+	}
+
+	failoverSummaryList := make([]*storkv1.FailoverSummary, 0)
+	scaleupStatus := true
+	for _, ns := range namespaces {
+		// scale replicas only if namespace is a subset of namespaces being migrated
+		logEvents := ac.printFunc(action, "ScaleReplicas")
+		logEvents(fmt.Sprintf("Scaling up apps in cluster %s", config.Host), "out")
+		var failoverSummary *storkv1.FailoverSummary
+		if slices.Contains(migrationNamespaces, ns) {
+			err := resourceutils.ScaleReplicasReturningError(ns, true, false, ac.printFunc(action, "ScaleReplicas"), config)
+			if err != nil {
+				scaleupStatus = false
+				msg := fmt.Sprintf("scaling up apps in namespace %s failed: %v", ns, err)
+				log.ActionLog(action).Errorf(msg)
+				failoverSummary = &storkv1.FailoverSummary{Namespace: ns, Status: storkv1.ActionStatusFailed, Reason: msg}
+			}
+			msg := fmt.Sprintf("scaling up apps in namespace %s successful", ns)
+			failoverSummary = &storkv1.FailoverSummary{Namespace: ns, Status: storkv1.ActionStatusSuccessful, Reason: msg}
```

Comment:
In case of error we are first updating failoverSummary with "failed" msg and later to successful msg.

Reply:
Thanks for pointing this out. I missed an else there. Will fix it as part of the revision
It should be 
```
if err != nil {
	scaleupStatus = false
	msg := fmt.Sprintf("scaling up apps in namespace %s failed: %v", ns, err)
	log.ActionLog(action).Errorf(msg)
	failoverSummary = &storkv1.FailoverSummary{Namespace: ns, Status: storkv1.ActionStatusFailed, Reason: msg}
} else {
        msg := fmt.Sprintf("scaling up apps in namespace %s successful", ns)
	failoverSummary = &storkv1.FailoverSummary{Namespace: ns, Status: storkv1.ActionStatusSuccessful, Reason: msg}
}
```